src/data/dfinsupp.lean

```diff
@@ -312,6 +312,28 @@ by simp only [single_apply, dif_neg h]
 lemma single_injective {i} : function.injective (single i : β i → Π₀ i, β i) :=
 λ x y H, congr_fun (mk_injective _ H) ⟨i, by simp⟩
 
+/-- Like `finsupp.single_eq_single_iff`, but with a `heq` due to dependent types -/
+lemma single_eq_single_iff (i j : ι) (xi : β i) (xj : β j) :
+  dfinsupp.single i xi = dfinsupp.single j xj ↔ i = j ∧ xi == xj ∨ xi = 0 ∧ xj = 0 :=
```

Comment:
I think I could shorten this to
```suggestion
  dfinsupp.single i xi = dfinsupp.single j xj ↔ xi == xj ∨ xi = 0 ∧ xj = 0 :=
```
but that makes it resemble `finsupp.single_eq_single_iff`, and probably makes it harder for the user to extract `i = j`

Reply:
Ah, of course I can't do this - otherwise when `β` is `λ i, X`, this would be false.